packages/flutter_tts/tizen/src/text_to_speech.h

```diff
@@ -11,30 +11,24 @@
 #include <string>
 #include <vector>
 
+enum class TtsState { kCreated, kReady, kPlaying, kPaused, kError };
```

Comment:
https://docs.tizen.org/application/native/api/mobile/5.5/group__CAPI__UIX__TTS__MODULE.html#ga3290b8dabf73cc17a1baac641465f28f

According to the document, there are no errors state.

Reply:
I believe it's implementing error-as-return value approach without using something like [StatusOr](https://abseil.io/docs/cpp/guides/status#overview-of-abslstatus). Similar implementation is done in [geolocator](https://github.com/flutter-tizen/plugins/blob/5671d565376eeca8a5701e96ffdf648a67a04bd4/packages/geolocator/tizen/src/permission_manager.h#L13-L19).

Reply:
My view is slightly different and I'd like to politely say that it doesn't matter that there's a precedent already merged. 

first, this enum and related method made to represent and to get TTS life-cycle on Tizen. therefore, using some state that does not exist on a state machine causes confusion. 

second, the return value of `tts_get_state` already represent the success of the function call and various error situations, 
so for error-as-return, It seems reasonable to use return value of `tts_get_state` as it is.

thirdly, furthermore, there seems to be a difference between using `StatusOr<T>` and not. This is similar to the `ErrorOr<T>` that we have been familiar. and the more generalized type is `std::expected` , and it has to be explicit like these. 
```
std::expected<TTsStatus, TizenError> TextToSpeech::GetState()
```

Reply:
As far as I know this style originated from the [battery_plus_tizen](https://github.com/flutter-tizen/plugins/blob/2bed8cbf8c2e9914531ff8db354e050f37c260a4/packages/battery_plus/tizen/src/device_battery.h#L14) and [connectivity_plus_tizen](https://github.com/flutter-tizen/plugins/blob/2bed8cbf8c2e9914531ff8db354e050f37c260a4/packages/connectivity_plus/tizen/src/connection.h#L20) plugins.

There's no reason for the values of `device_battery_status_e` and `BatteryStatus` to exactly match with each other's. They are just different types by design.

```cpp
enum class BatteryStatus { kFull, kCharging, kDischarging, kUnknown, kError };

typedef enum {
	DEVICE_BATTERY_STATUS_CHARGING, /**< Battery is charging */
	DEVICE_BATTERY_STATUS_DISCHARGING, /**< Battery is discharging */
	DEVICE_BATTERY_STATUS_FULL, /**< Battery is fully charged */
	DEVICE_BATTERY_STATUS_NOT_CHARGING, /**< Battery is not charging */
} device_battery_status_e;
```

The user of the `DeviceBattery` class should only care about the public interface of the class but not its underlying implementation.

Reply:
That's right, the two types don't necessarily have to be symmetrical.

However, IMO, I'd like to politely say that the error is not the status. the error mentioned here only indicates that it was interrupted by an error during a function call. The same goes for the battery status. The current definitions is ambiguous and confuses those who use the API. In addition, it is not fully documented.

Reply:
Well, an error status is also a possible result of a function and I think everyone can understand what an `kError` state is without difficulty.

Reply:
Of course it can be. but when newcomers try to use it, he can feel the ambiguity. because he doesn't know if this is a function call failure or if this module can actually go into an error state.

I think there is nothing wrong with keeping to use this style in terms of program behavior.
In that case, however, additional comments are required.

Reply:
I agree that the meaning of Error in State is ambiguous. And I don't think we have to necessarily follow the definition of tts_state_e also.
But in this case, as per @bbrto21's comment. I think using std::optional can make it a more clearer.
If you agree with this method, I will try to apply it to other code.

Reply:
@JSUYA Thank you but you don't need to propagate this to every plugin :)